pkg/controller/operators/catalog/operator.go

```diff
@@ -1181,6 +1181,7 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 	// TODO: parallel
 	maxGeneration := 0
 	subscriptionUpdated := false
+	subsSet := sets.New[string]()
```

Comment:
Could we make this a set of a [`types.NamespacedName`](https://pkg.go.dev/k8s.io/apimachinery/pkg/types#NamespacedName) instead?
```suggestion
	subsSet := sets.New[types.NamespacedName]()
```

That way we don't have to worry about string concatenation resulting in inadvertent key reuse. E.g.
- `namepace: foo, name: bar-bazz` => `key: foo-bar-bazz`
- `namespace: foo-bar, name: bazz` => `key: foo-bar-bazz`